include/deal.II/fe/fe_values.h

```diff
@@ -2175,9 +2177,24 @@ class FEValuesBase : public Subscriptor
   static const unsigned int space_dimension = spacedim;
 
   /**
-   * Number of quadrature points.
+   * Number of quadrature points of the current object. Its value is
+   * initialized by the value of max_quadrature_points and is updated,
+   * e.g., if FEFaceValues::reinit() is called for a new cell/face.
+   *
+   * @note The default value equals to the value of max_quadrature_points.
    */
-  const unsigned int n_quadrature_points;
+  unsigned int n_quadrature_points;
```

Comment:
This is a potentially dangerous change, given that it is a public member and non-const. Now I know we won't have a clean solution here, but I would prefer (if possible) to keep this variable `const` and use `const_cast` at the single position this variable can change. What do others think?

Reply:
We should make such variables private^^

Reply:
But that variables are public is an inherent problem in deal.II. See also `FiniteElementData`: `dofs_per_face` vs. `n_dofs_per_face()` with the latter only working for wedges/pyramids.

Reply:
I agree. The other possibility would be to always return the value 0 and have a getter function for the proper content.

Reply:
I have made the change (`const_cast`). But this really hurts and I makes me doubt how the compiler can make any assumptions about variables.

Reply:
I think it is OK and defined behavior because the `this` pointer is not constant in this context. It would be different if we were messing with a `const` object or even `constexpr` because then the compiler might never create a storage location for that variable.